components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/CertificateMgtDAO.java

```diff
@@ -237,32 +225,26 @@ private CertificateMetadataDTO getCertificate(String alias, String endpoint) thr
         String getCertQuery = SQLConstants.CertificateConstants.GET_CERTIFICATE_ALL_TENANTS;
 
         try {
-            try {
-                connection = APIMgtDBUtil.getConnection();
-                initialAutoCommit = connection.getAutoCommit();
-                connection.setAutoCommit(false);
-                connection.commit();
-                preparedStatement = connection.prepareStatement(getCertQuery);
-                preparedStatement.setString(1, alias);
-                preparedStatement.setString(2, endpoint);
-                resultSet = preparedStatement.executeQuery();
+            connection = APIMgtDBUtil.getConnection();
+            initialAutoCommit = connection.getAutoCommit();
+            connection.setAutoCommit(false);
+            connection.commit();
+            preparedStatement = connection.prepareStatement(getCertQuery);
+            preparedStatement.setString(1, alias);
+            preparedStatement.setString(2, endpoint);
+            resultSet = preparedStatement.executeQuery();
 
-                while (resultSet.next()) {
-                    certificateMetadataDTO = new CertificateMetadataDTO();
-                    certificateMetadataDTO.setAlias(resultSet.getString("ALIAS"));
-                    certificateMetadataDTO.setEndpoint(resultSet.getString("END_POINT"));
-                }
-            } catch (SQLException e) {
-                handleException("Error while retrieving certificate metadata.", e);
-            } finally {
-                connection.setAutoCommit(initialAutoCommit);
-                APIMgtDBUtil.closeStatement(preparedStatement);
-                APIMgtDBUtil.closeAllConnections(preparedStatement, connection, resultSet);
+            while (resultSet.next()) {
+                certificateMetadataDTO = new CertificateMetadataDTO();
+                certificateMetadataDTO.setAlias(resultSet.getString("ALIAS"));
+                certificateMetadataDTO.setEndpoint(resultSet.getString("END_POINT"));
             }
         } catch (SQLException e) {
-            if (log.isDebugEnabled()) {
-                log.debug("Error occurred in finally block while retrieving certificate metadata. ", e);
-            }
+            handleException("Error while retrieving certificate metadata.", e);
+        } finally {
+            APIMgtDBUtil.setAutoCommit(connection, initialAutoCommit);
+            APIMgtDBUtil.closeStatement(preparedStatement);
+            APIMgtDBUtil.closeAllConnections(preparedStatement, connection, resultSet);
```

Comment:
Do we need to explicitly call APIMgtDBUtil.closeStatement? APIMgtDBUtil.closeAllConnections does that as well.